actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/qrcodedownload/QRCodeDownloadManagementActor.java

```diff
@@ -0,0 +1,223 @@
+                    ResponseCode.errorUserHasNotCreatedAnyCourse.getErrorMessage(),
+                    ResponseCode.CLIENT_ERROR.getResponseCode());
+        Map<String, List<String>> dialCodesMap = contents.stream()
+                .filter(content -> content.get("dialcodes") != null)
+                .filter(content -> content.get("name") != null)
+                .collect(Collectors.toMap(content -> ((String) content.get("identifier")) + "<<<" + (String) content.get("name"), content -> (List) content.get("dialcodes")));
+        File file = generateCSVFile(dialCodesMap);
+        Response response = new Response();
+        if (null == file)
+            throw new ProjectCommonException(
+                    ResponseCode.errorProcessingFile.getErrorCode(),
+                    ResponseCode.errorProcessingFile.getErrorMessage(),
+                    ResponseCode.SERVER_ERROR.getResponseCode());
+
+        response = uploadToAws(file);
+        response.put(JsonKey.USER_IDs, requestMap.get("userIds"));
+        sender().tell(response, self());
+    }
+
+    /**
+     * Search call to LP composite search engine
+     * @param requestMap
+     * @param headers
+     * @return
+     */
+    private Map<String, Object> searchCourses(Map<String, Object> requestMap, Map<String, String> headers) {
+        String request = prepareSearchRequest(requestMap);
+        Map<String, Object> searchResponse = ContentSearchUtil.searchContentSync(null, request, headers);
+        return searchResponse;
+    }
+
+    /**
+     * Request Preparation for search Request
+     * @param requestMap
+     * @return
+     */
+    private String prepareSearchRequest(Map<String, Object> requestMap) {
+        Map<String, Object> searchRequestMap = new HashMap<String, Object>() {{
+            put(JsonKey.FILTERS, requestMap.keySet().stream()
+                    .filter(key -> filtersHelperMap.containsKey(key))
+                    .collect(Collectors.toMap(key -> filtersHelperMap.get(key), key -> requestMap.get(key))));
+            put(JsonKey.FIELDS, fields);
+            put(JsonKey.LIMIT, 200);
+        }};
+        Map<String, Object> request = new HashMap<String, Object>() {{
+            put(JsonKey.REQUEST, searchRequestMap);
+        }};
+        String requestJson = null;
+        try {
+            requestJson = new ObjectMapper().writeValueAsString(request);
+        } catch (JsonProcessingException e) {
+            ProjectLogger.log("QRCodeDownloadManagement:prepareSearchRequest: Exception occurred with error message = " + e.getMessage(), e);
+        }
+        return requestJson;
+    }
+
+    /**
+     * Generates the CSV File for the data provided
+     * @param dialCodeMap
+     * @return
+     */
+    private File generateCSVFile(Map<String, List<String>> dialCodeMap) {
+        File file = null;
+        if (MapUtils.isNotEmpty(dialCodeMap)) {
+            try {
+                file = new File(UUID.randomUUID().toString() + ".csv");
```

Comment:
Is it good to add a random UUID as fileName, how do we know it is for a particular dialcode of a course

Reply:
As per requirement, I think UUID itself is required.